cfbs/commands.py

```diff
@@ -201,15 +202,96 @@ def local_module_copy(module, counter, max_length):
         f"{counter:03d} {pad_right(name, max_length)} @ local                                    (Copied)"
     )
 
+def _get_path_from_url(url):
+    if not url.startswith(("https://", "ssh://", "git://")):
+        if ("://" in url):
+            return user_error("Unsupported URL protocol in '%s'" % url)
+        else:
+            # It's a path already, just remove trailing slashes (if any).
+            return url.strip("/")
+
+    path = None
+    if url.startswith("ssh://"):
+        match = re.match(r"ssh://(\w+)@(.+)", url)
+        if match is not None:
+            path = match[2]
+    path = path or url[url.index("://") + 3:]
+    path = strip_right(path, ".git")
+    path = path.strip("/")
+
+    return path
+
+def _get_git_repo_commit_sha(repo_path):
+    assert os.path.isdir(os.path.join(repo_path, ".git"))
+
+    with open(os.path.join(repo_path, ".git", "HEAD"), "r") as f:
+        head_ref_info = f.read()
+
+    assert head_ref_info.startswith("ref: ")
+    head_ref = head_ref_info[5:].strip()
+
+    with open(os.path.join(repo_path, ".git", head_ref)) as f:
+        return f.read().strip()
+
+def _clone_index_repo(repo_url):
+    assert repo_url.startswith(("https://", "ssh://", "git://"))
+
+    commit = None
+    if "@" in repo_url and (repo_url.rindex("@") > repo_url.rindex(".")):
```

Comment:
Well, I think `@some-branch` would just work, right? But I don't quite want to support it because it might give people impression `cfbs build` would fetch/rebase from the branch. Better let them think they should use a SHA. :)

Reply:
This is a valid point and I will create some tickets around it, starting with this one:

https://tracker.mender.io/browse/CFE-3797

(I'll add some more tickets to same epic).

No need to do anything more in this PR.